util/topsql/reporter/reporter.go

```diff
@@ -241,8 +302,15 @@ func (tsr *RemoteTopSQLReporter) Collect(timestamp uint64, records []tracecpu.SQ
 
 // Close uses to close and release the reporter resource.
 func (tsr *RemoteTopSQLReporter) Close() {
+	tsr.dataSinks.mu.Lock()
+	defer tsr.dataSinks.mu.Unlock()
+
+	for d := range tsr.dataSinks.m {
+		d.CloseDataSink()
+		delete(tsr.dataSinks.m, d)
+	}
```

Comment:
It is possible to avoid the possibility of deadlock by reducing the lock scope to where it is just needed. Steps as below:

1. Take out the entire map with a lock protected.
2. Iterate the map and close data sinks one by one, without locking.